api/src/main/java/org/openmrs/module/shr/odd/generator/section/impl/AllergiesSectionGenerator.java

```diff
@@ -271,7 +252,33 @@ else if(this.getSectionObs().size() > 0) // unstructured?
 
 		return retVal;
 	}
-	
+
+	private Obs findLastProblemObs(Allergy allergy){
+		List<Obs> candidates= new ArrayList<>();
+		List<Obs> obs = Context.getObsService().getObservationsByPerson(allergy.getPatient());
+		for(Obs currentObs : obs){
+			if(currentObs.getValueCoded() != null && currentObs.getValueCoded().equals(allergy.getAllergen().getCodedAllergen()))
```

Comment:
We should use one of the methods from ObsService to search more specifically using the concept:

https://github.com/openmrs/openmrs-core/blob/master/api/src/main/java/org/openmrs/api/ObsService.java (crtl+f valuecoded)

This code can be a performance issue - it's better to use the methods that forward these searches to the db.